packages/hoppscotch-common/src/newstore/localpersistence.ts

```diff
@@ -210,35 +208,21 @@ function setupEnvironmentsPersistence() {
 }
 
 function setupSelectedEnvPersistence() {
```

Comment:
when dealing with localstorage values, backward compatibility is important because different versions of the app may have different values.

in this case our old app can have selectedEnvIndex as -1, which is not a valid `selectedEnvIndex` in our new implementation. so we need to check for that handle it properly. otherwise our app can break.

there is a function `checkAndMigrateOldSettings`, where you can add the check. 

```suggestion
const selectedEnvIndex = localStorage.getItem("selectedEnvIndex")

  if (selectedEnvIndex == "-1") {
    localStorage.setItem(
      "selectedEnvIndex",
      JSON.stringify({
        type: "NO_ENV_SELECTED",
      })
    )
  }
```